keps/sig-storage/1710-selinux-relabeling/README.md

```diff
@@ -346,59 +541,54 @@ To mitigate this risk, we propose:
-   present in DSW. This should cover only volumes that don't have a Pod in the API server and need to be unmounted. Kubelet adds the
-   volumes to the ASW and lets regular reconciler to unmount them.
-
-This approach does not work for SELinux, because at step 1. above, the volume manager needs to know *if* a volume is mounted and with
-*what SELinux context mount option*. If the required and existing SELinux contexts of a volume match, the volume manager can continue
-mounting the volume. If they don't, volume manager needs to unmount the volume with the wrong SELinux context first and mount it again
-with the right one.
-
-We need to populate the ASW as soon as possible after kubelet starts. Suggested changes:
-
-1. When kubelet starts, the volume manager will reconstruct all volumes incl. their SELinux contexts and put them to the DSW as *uncertain*.
-   At this point, kubelet may not have connection to the API server yet, hence this phase of volume reconstruction must work without it.
-   Kubelet will store all reconstructed volumes in a separate array, to finish the reconstruction when the API server is available.
-   * This implies that volume plugins can't expect that the API server is available in `ConstructVolumeSpec`, `ConstructBlockVolumeSpec`,
-     `NewMounter`, `NewBlockVolumeMapper`, and `NewDeviceMounter` calls. Especially all `CSIDriver` checks in the CSI volume plugin must
-     be moved to `SetUpAt` or `TearDownAt`, and their block volume counterparts.
-2. Only after the initial ASW is populated, kubelet starts running pods and mounting volumes for them. Since the existing volumes are marked
-   as *uncertain*, volume manager will re-mount them (depending on volume plugin / CSI driver idempotency). Note that only mounting
-   is allowed at this point, the volume manager can't unmount anything, because the DSW is not yet populated.
-3. When kubelet establishes a connection to the API server, it populates the DSW as usual.
-4. When the DSW is fully populated, the volume manager will finish reconstruction of volumes, i.e. file devicePaths from the
-   `node.status.volumesInUse` field.
-5. Only after the second phase of volume reconstruction is done, i.e. the DSW is fully populated and volumes are fully reconstructed,
-   the volume manager starts unmounting volumes that are not in the ASW.
-
-This work was separated into its own KEP + Feature [#3756](https://github.com/kubernetes/enhancements/issues/3756).
-
-Here in this KEP we will need only to add SELinux context to reconstructed volumes.
+This work was separated into its own KEP + Feature [#3756](https://github.com/kubernetes/enhancements/issues/3756) and went GA in Kubernetes 1.30.
+
+Here in this KEP we will need only to add SELinux label to reconstructed volumes.
+
+### kube-state-metrics
+
+*This section is under development. We need to figure out is Prometheus is able to do such complex queries.*
```

Comment:
What would be the alternative if Prometheus can't handle it (which is going to be hard to determine)? Should the kubelet emit a gauge metric per container and/or pod instead?